aether-kernel/aether/kernel/api/migrations/0102_auto_20190405_0737.py

```diff
@@ -0,0 +1,107 @@
+# Generated by Django 2.1.8 on 2019-04-05 07:37
+
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+import model_utils.fields
+import uuid
+
+def migrate_ProjectSchema_to_SchemaDecorator(apps, schema_editor):
+    ProjectSchemas = apps.get_model('kernel', 'ProjectSchema')
+    SchemaDecorators = apps.get_model('kernel', 'SchemaDecorator')
+    Mappings = apps.get_model('kernel', 'Mapping')
+    Entities = apps.get_model('kernel', 'Entity')
+
+    for projectschema in ProjectSchemas.objects.all():
+        SchemaDecorators.objects.create(
+            name=projectschema.name,
+            project=projectschema.project,
+            schema=projectschema.schema,
+            mandatory_fields=projectschema.mandatory_fields,
+            transport_rule=projectschema.transport_rule,
+            masked_fields=projectschema.masked_fields,
+            is_encrypted=projectschema.is_encrypted,
+            id=projectschema.id,
+        )
+
+    for mapping in Mappings.objects.all():
+        for ps in mapping.projectschemas.all():
+            mapping.schemadecorators.add(str(ps.pk))
+
+    for entity in Entities.objects.all():
+        entity.schemadecorator = entity.projectschema
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('kernel', '0101_name_not_unique'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
```

Comment:
Instead of `CreateModel`, there is one more option `RenameModel`  https://docs.djangoproject.com/en/2.2/ref/migration-operations/#renamemodel. With that you skip all the manual updates.

Also :
- https://docs.djangoproject.com/en/2.2/ref/migration-operations/#renamefield (rename all `projectschema` fields)
- https://docs.djangoproject.com/en/2.2/ref/migration-operations/#altermodeltable (to change the table name `db_table`)

Reply:
@obdulia-losantos this approach will require that the bridge table crated on the db for `Mapping > schemadecorators` be modified outside django or we will have to define the new bridge table using the `through` field property. I think this is an overkill.

Reply:
let me finish what I'm doing now and I'll try it :crossed_fingers:

Reply:
Now, I'm trying to check this with the rename option :crossed_fingers: 

Another thing is if you could rename the migration file to `0102_schema_decorator.py` or something like that... :grimacing: